table sort/active tab conflict

mantis-toboggan-md opened the pull request:
#2327 - This issue was happening because both Tabs and SortableTable were using route query params to track their state state (active tab, table sort order and page). Updating SortableTable to track column sorting and page number only in local state (`this.sortBy this.descending; this.currentPage`) instead of query params avoids the problem.

codyrancher reviewed (approved):
I'm okay with these changes but it looks like it will break being able to share pre-sorted tables. 

I don't think it's a feature we need. @vincent99 is there a special case you want to support?

vincent99 commented:
These and other query params exist so that (as much as practical) if you're on a page and hit reload you get the same view back again.  I would like to maintain that.  It's both useful on it's own and particularly helpful if you're developing something in the 3rd tab of a view page or something.

They actually **don't** both use query params, which is really the problem.. the tabs uses the [window.location.]hash, and the hash is not preserved/copied into the sort links when generated by `:to={query: { the new query param for this sort}}`.  Which makes sense from the router's perspective since if you're linking to another page you wouldn't normally want to preserve the current page anchor.

So switching the tabs to keep the active tab as a query param instead of in the hash (`blah?tab=ports` instead of `blah#ports`) would allow both to work together (query params not touched by the :to are preserved).

vincent99 reviewed (changes requested):
^--

mantis-toboggan-md commented:
I updated the table headers to use applyQuery instead of :to="{query}" which preserves the hash. 2 other small fixes: 
*  Changing the sort order of tables in resource detail views causes the whole detail component to re-render: I updated ResourceDetail to only re-run fetch when "mode" or "as" params change to avoid this
* updated the paging logic to only call applyQuery when the page number actually changes, rather than every time sort order changes (which occasionally causes navigation cancelled errors)
